src/purple/blist.rs

```diff
@@ -0,0 +1,40 @@
+use super::ffi::AsPtr;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ptr::NonNull;
+
+pub struct BlistNode(NonNull<purple_sys::PurpleBlistNode>);
+
+impl BlistNode {
+    pub unsafe fn from_ptr(ptr: *mut purple_sys::PurpleBlistNode) -> Option<Self> {
+        NonNull::new(ptr).map(Self)
+    }
+
+    pub fn get_string(&mut self, c_key: &CString) -> Option<String> {
```

Comment:
It could take a `&str`. I usually only use `CString` when somehow the caller must manage the string memory or we need to ensure we use `&'static CStr`. But in this case, `purple_blist_node_(get|set)_string` does not take ownership of the string so we can create a temporary `CString` like we do elsewhere.

Reply:
done!